src/libs/actions/TaxRate.ts

```diff
@@ -119,10 +182,299 @@ function clearTaxRateError(policyID: string, taxID: string, pendingAction?: Pend
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesToDelete.reduce<TaxRateDeleteMap>((acc, taxID) => {
+                            acc[taxID] = null;
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesToDelete.reduce<TaxRateDeleteMap>((acc, taxID) => {
+                            acc[taxID] = {
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+                                errors: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.deleteFailureMessage'),
+                            };
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        taxNames: JSON.stringify(taxesToDelete.map((taxID) => policyTaxRates[taxID].name)),
+    } satisfies DeletePolicyTaxesParams;
+
+    API.write(WRITE_COMMANDS.DELETE_POLICY_TAXES, parameters, onyxData);
+}
+
+/**
+ * Rename policy tax
+ */
+function updatePolicyTaxValue(policyID: string, taxID: string, taxValue: number) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+    const originalTaxRate = {...policy?.taxRates?.taxes[taxID]};
+    const stringTaxValue = `${taxValue}%`;
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {
+                                value: stringTaxValue,
+                                pendingFields: {value: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+                                errorFields: {value: null},
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {value: stringTaxValue, pendingFields: {value: null}, errorFields: {value: null}},
```

Comment:
Same here, no need to update the `value` field, we already did that in `optimisticData`
```suggestion
                            [taxID]: {pendingFields: {value: null}, errorFields: {value: null}},
```